app/models/payment/braintree/transaction.rb

```diff
@@ -45,4 +47,8 @@ def publish_subscription_charge
     },
                         { group_id: "braintree-subscription:#{subscription.id}" })
   end
+
+  def increment_funding_counter
+    FundingCounter.update(page: page, currency: currency, amount: amount)
```

Comment:
What's the benefit of the service vs having the logic in the model? In the model we could do a `transactions.sum(:amount_in_usd)` (granted we added an `amount_in_usd` column) and it would also allow us to check what was refunded. (That aso brings to mind that we might need an `after_update`?)

Reply:
I wanted to extract it into the service instead of duplicating the exact same code (with what back then also needed a private helper method which I refactored out) in both the GoCardless and the Braintree Transaction model. I'd prefer to not add another migration to add another column to the transactions table. Can you tell me what you mean that it'd allow us to check was refunded? FWIW, we also don't purge actions from the actions tally when someone asks for their data to be anonymized - I think these counters are supposed to be robust and approximate.